### PR TITLE
Smart Proxy (and core) for ACD

### PR DESCRIPTION
**Pull request by sbernhard**

### CODE REVIEW

File: comps/comps-foreman-plugins-rhel7.xml [hunk 1]
```diff
@@ -172,7 +173,9 @@
       <!--
         Do not edit this section manually and use ./comps_doc.sh script
       -->
+      <packagereq type="default">rubygem-smart_proxy_acd</packagereq>
       <!--RGD-START-->
+      <packagereq type="default">rubygem-smart_proxy_acd-doc</packagereq>
```
Comment (evgeni) [inline comps/comps-foreman-plugins-rhel7.xml:178]: the smart proxy is SCL'ed these days too, so it should be `tfm-rubygem-…` and then also sorted accordingly ;)

File: comps/comps-foreman-el8.xml [hunk 2]
```diff
@@ -410,8 +410,8 @@
       <packagereq type="default">rubygem-gitlab-sidekiq-fetcher-doc</packagereq>
       <packagereq type="default">rubygem-globalid-doc</packagereq>
       <packagereq type="default">rubygem-google-api-client-doc</packagereq>
-      <packagereq type="default">rubygem-google-cloud-env-doc</packagereq>
```
Comment (evgeni) [inline comps/comps-foreman-el8.xml:413]: imho this change is noise :)

File: comps/comps-foreman-rhel7.xml [hunk 3]
```diff
@@ -434,8 +434,8 @@
       <packagereq type="default">tfm-rubygem-gitlab-sidekiq-fetcher-doc</packagereq>
       <packagereq type="default">tfm-rubygem-globalid-doc</packagereq>
       <packagereq type="default">tfm-rubygem-google-api-client-doc</packagereq>
-      <packagereq type="default">tfm-rubygem-google-cloud-env-doc</packagereq>
```
Comment (evgeni) [inline comps/comps-foreman-rhel7.xml:437]: imho this change is noise :)

File: package_manifest.yaml [hunk 4]
```diff
@@ -1087,6 +1088,7 @@ plugin_nonscl_packages:
     ansiblerole-satellite-receptor-installer: {}
     puppet-foreman_scap_client: {}
     puppetlabs-stdlib: {}
+    rubygem-smart_proxy_acd: {}
```
Comment (evgeni) [inline package_manifest.yaml:1091]: this should be in `plugin_scl_packages` -- does `add_gem_package` need an update here?

File: packages/plugins/rubygem-smart_proxy_acd/rubygem-smart_proxy_acd.spec [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+
+%{!?_root_bindir:%global _root_bindir %{_bindir}}
+%{!?_root_datadir:%global _root_datadir %{_datadir}}
+%{!?_root_localstatedir:%global _root_localstatedir %{_localstatedir}}
+%{!?_root_sysconfdir:%global _root_sysconfdir %{_sysconfdir}}
+
+%global gem_name smart_proxy_acd
+%global plugin_name acd
+
+%global foreman_proxy_min_version 1.25
```
Comment (evgeni) [inline packages/plugins/rubygem-smart_proxy_acd/rubygem-smart_proxy_acd.spec:13]: there is no such thing as 1.25 -- did you mean 2.0?

File: packages/plugins/rubygem-smart_proxy_acd/rubygem-smart_proxy_acd.spec [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+URL: http://github.com/ATIX-AG/smart_proxy_acd
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+%if 0%{?rhel} == 7
+Requires: tfm-rubygem(smart_proxy_dynflow_core) >= 0.2.2
+Requires: tfm-rubygem(smart_proxy_acd_core)
+%else
+Requires: rubygem(smart_proxy_dynflow_core) >= 0.2.2
+Requires: rubygem(smart_proxy_acd_core)
+%endif
```
Comment (evgeni) [inline packages/plugins/rubygem-smart_proxy_acd/rubygem-smart_proxy_acd.spec:35]:
```suggestion
Requires: %{?scl_prefix}rubygem(smart_proxy_dynflow_core) >= 0.2.2
Requires: %{?scl_prefix}rubygem(smart_proxy_acd_core)
```

File: packages/plugins/rubygem-smart_proxy_acd/rubygem-smart_proxy_acd.spec [hunk 7]
```diff
@@ -0,0 +1,130 @@
...
+Requires: %{?scl_prefix_ruby}ruby(rubygems)
+BuildRequires: %{?scl_prefix_ruby}ruby(release)
+BuildRequires: %{?scl_prefix_ruby}ruby
+BuildRequires: %{?scl_prefix_ruby}rubygems-devel
+BuildArch: noarch
+Provides: %{?scl_prefix}rubygem(%{gem_name}) = %{version}
+Provides: foreman-proxy-plugin-%{plugin_name} = %{version}
+# end specfile generated dependencies
+
+%{?scl:Obsoletes: rubygem-%{gem_name}}
```
Comment (evgeni) [inline packages/plugins/rubygem-smart_proxy_acd/rubygem-smart_proxy_acd.spec:50]: not required for a new plugin

File: rel-eng/tito.props [hunk 8]
```diff
@@ -951,6 +952,7 @@ whitelist = ansiblerole-foreman_scap_client
   ansiblerole-insights-client
   ansiblerole-satellite-receptor-installer
   puppet-foreman_scap_client
+  rubygem-smart_proxy_acd
```
Comment (evgeni) [inline rel-eng/tito.props:955]: needs to go into scl

**Review by evgeni — CHANGES_REQUESTED** [review]
this needs a rebase for latest changes to package_manifest.yml

also the package needs adding to the EL8 configs (comps, tito.props) -- our scripts currently don't do that automatically :(

also, some inline comments about the proper spec layout for SCL'ed proxy

**Comment by sbernhard** [comment]
thank you very much @evgeni for your help! let me know if its ok now.

**Review by evgeni — APPROVED** [review]

**Comment by evgeni** [comment]
@sbernhard if you want that for 2.4, please cherrypick :)

**Pull request merged**